GetStream/stream-chat-android
# [CAS-1089] Fix crash when adding reactions in threads

https://stream-io.atlassian.net/browse/CAS-1089

### 🎯 Goal

Fix crash when adding reactions in threads

### 🛠 Implementation details

For some reason, we don't do caching when querying for messages in threads. Therefore, foreign key constraint for added reactions is not satisfied. The two possible approaches are:
1. Implement proper caching in threads
2. Don't store reactions via `ReactionsRepository` when adding a reaction in threads

This PR implements the second approach.

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/9600921/122554364-8fdcdc80-d041-11eb-8471-20fc249ae106.mov" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/9600921/122554263-6fad1d80-d041-11eb-8601-df2832377d45.mov" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

Steps to reproduce:
1. Sing in as user 1
2. Send a thread message
3. Sign in as user 2
4. Leave a reaction on the very same thread message

Expected result: 
The message contains 2 reactions: one reaction from user 1 and another one from user 2

Actual result:
Crash

### ☑️ Checklist


### 🎉 GIF

![giphy (2)](https://user-images.githubusercontent.com/9600921/122554810-1ee9f480-d042-11eb-9e10-95e50f4f0fc5.gif)
